module/apmawssdkgo/sns.go

```diff
@@ -0,0 +1,119 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmawssdkgo // import "go.elastic.co/apm/module/apmawssdkgo"
+
+import (
+	"strings"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmhttp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/sns"
+)
+
+type apmSNS struct {
+	name, opName, resourceName string
+}
+
+func newSNS(req *request.Request) (*apmSNS, error) {
+	if req.Operation.Name != "Publish" {
+		return nil, errMethodNotSupported
+	}
+	name := req.ClientInfo.ServiceID + " PUBLISH"
+	resourceName := serviceSNS
+
+	topicName := getTopicName(req)
+	if topicName != "" {
+		name += " " + topicName
+		resourceName += "/" + topicName
+	}
+
+	s := &apmSNS{
+		name:         name,
+		opName:       "publish",
+		resourceName: resourceName,
+	}
+
+	return s, nil
+}
+
+func (s *apmSNS) spanName() string { return s.name }
+
+func (s *apmSNS) resource() string { return s.resourceName }
+
+func (s *apmSNS) setAdditional(span *apm.Span) {
+	span.Action = s.opName
+}
+
+func getTopicName(req *request.Request) string {
+	// TODO: PhoneNumber is the third possibility, but I'm guessing we
+	// don't want to store that for customers?
```

Comment:
I figure we don't want to store something with such explosive cardinality, but made a note in case you know anything else. 

@AlexanderWert do you know?